src/FakeItEasy/Configuration/BuildableCallRule.cs

```diff
@@ -160,6 +160,24 @@ public virtual void ApplyWherePredicate(Func<IFakeObjectCall, bool> predicate, A
 
         public abstract void UsePredicateToValidateArguments(Func<ArgumentCollection, bool> predicate);
 
+        /// <summary>
+        /// Clones the part of the rule that describes which call is being configured.
+        /// </summary>
+        /// <returns>The cloned rule.</returns>
+        public BuildableCallRule CloneCallSpecification()
+        {
+            var clone = this.CloneCallSpecificationCore();
+            clone.wherePredicates.AddRange(this.wherePredicates);
+            return clone;
+        }
+
+        /// <summary>
+        /// When overridden in a derived class, returns a new instance of the same type and copies the call
+        /// specification members for that type.
+        /// </summary>
+        /// <returns>The cloned rule.</returns>
+        protected abstract BuildableCallRule CloneCallSpecificationCore();

```

Comment:
I can't stop myself from detecting a code smell here:

- Only the docs are telling an implementer that they should return the same type as the implementer. Should there be a generic type param enforcing this instead?
- The "core" suffix. Is there a name that could better express the intent of this method and distinguish it from `CloneCallSpecification`?

That said, the type is internal so it's not of utmost importance to address these concerns.